lib/state_machine/state_machine.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "state.cpp"
+#include "transition.cpp"
+
+namespace hyped::state_machine {
+
+class StateMachine {
+ public:
+  StateMachine();
+
+  void transition(Message message);
+
+  void checkTransition();
+
+  void reset();
+
+  // TODOLater change states to actual states
+  State states[5]
+    = {State::kInitialState, State::state1, State::state2, State::state3, State::state4};
+
+  // TODOLater change transitions to actual transitions
+  Transition transitions[5] = {
```

Comment:
You've implicitly used the number of states a few times now without stating where the number 5 comes from. Better design would be to create a variable somewhere (e.g. `kNumStates = 5`) and refer to that?